src/main/java/org/jenkinsci/plugins/githubautostatus/notifiers/StatsdNotifier.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 jxpearce.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.githubautostatus.notifiers;
+
+import org.jenkinsci.plugins.githubautostatus.StatsdWrapper;
+import org.jenkinsci.plugins.githubautostatus.StatsdNotifierConfig;
+
+import java.util.logging.Logger;
+
+/**
+ * Sends job and stage metrics to a statsd collector server over UDP.
+ * @author Tom Hadlaw (thomas.hadlaw@hootsuite.com)
+ */
+public class StatsdNotifier implements BuildNotifier {
+    private StatsdWrapper client;
+    protected StatsdNotifierConfig config;
+    private static final Logger LOGGER = Logger.getLogger(StatsdWrapper.class.getName());
+
+    public StatsdNotifier(StatsdNotifierConfig config) {
+        this.config = config;
+        int port = 8125;
+        if (!config.getStatsdPort().equals("")) {
+            try {
+                port = Integer.parseInt(config.getStatsdPort());
```

Comment:
Not sure if it matters but this wouldn't catch ports outside of port range. E.g. above 65535 or 0